src/utils/cache.ts

```diff
@@ -24,18 +25,30 @@ export function canCache(method?: string): boolean {
   return cacheTypeForMethod(method) !== 'never';
 }
 
-export function blockTagForPayload(payload: Payload): string | undefined {
-  if (!payload.params) {
+/**
+ * Return the block parameter for the given request, if it has one.
+ *
+ * @param request - The JSON-RPC request.
+ * @returns The block parameter in the given request, or `undefined` if none was found.
+ */
+export function blockTagForRequest(
+  request: JsonRpcRequest<unknown>,
+): string | undefined {
```

Comment:
Hmm. How does TypeScript know that the block param will be a string if the params type parameter is `unknown`?

I'm wondering if it makes more sense to do:

``` suggestion
export function blockTagForRequest<T>(
  request: JsonRpcRequest<unknown>,
): T | undefined {
```

or even:

``` suggestion
export function blockTagForRequest(
  request: JsonRpcRequest<unknown>,
): unknown | undefined {
```

Reply:
.... huh. That's a great point. Not sure why it let me do that.

Reply:
Wow, apparently TypeScript will let me set this left type to anything! No errors. That seems like a bug.